src/main/java/org/embulk/input/jira/JiraInputPlugin.java

```diff
@@ -0,0 +1,233 @@
+            Schema schema, int taskIndex,
+            PageOutput output)
+    {
+        PluginTask task = taskSource.loadTask(PluginTask.class);
+        JiraUtil.validateTaskConfig(task);
+        JiraClient jiraClient = getJiraClient();
+        jiraClient.checkUserCredentials(task);
+        try (final PageBuilder pageBuilder = getPageBuilder(schema, output)) {
+            if (isPreview()) {
+                List<Issue> issues = jiraClient.searchIssues(task, 0, PREVIEW_RECORDS_COUNT);
+                for (Issue issue : issues) {
+                    issue.toRecord();
+                    JiraUtil.addRecord(issue, schema, task, pageBuilder);
+                }
+            }
+            else {
+                int currentPage = 0;
+                int totalCount = jiraClient.getTotalCount(task);
+                int totalPage = JiraUtil.calculateTotalPage(totalCount, MAX_RESULTS);
+                LOGGER.info(String.format("Total pages (%d)", totalPage));
+                while (currentPage < totalPage) {
+                    LOGGER.info(String.format("Fetching page %d/%d", (currentPage + 1), totalPage));
+                    List<Issue> issues = jiraClient.searchIssues(task, (currentPage * MAX_RESULTS), MAX_RESULTS);
+                    for (Issue issue : issues) {
+                        issue.toRecord();
+                        JiraUtil.addRecord(issue, schema, task, pageBuilder);
+                    }
+                    currentPage++;
+                }
+            }
+            pageBuilder.finish();
+        }
+        return Exec.newTaskReport();
+    }
+
+    @Override
+    public ConfigDiff guess(ConfigSource config)
+    {
+        // Reset columns in case already have or missing on configuration
+        config.set("columns", new ObjectMapper().createArrayNode());
+        ConfigSource guessConfig = createGuessConfig();
+        GuessExecutor guessExecutor = getGuessExecutor();
+        PluginTask task = config.loadConfig(PluginTask.class);
+        JiraUtil.validateTaskConfig(task);
+        JiraClient jiraClient = getJiraClient();
+        jiraClient.checkUserCredentials(task);
+        List<Issue> issues = jiraClient.searchIssues(task, 0, GUESS_RECORDS_COUNT);
+        issues.forEach(Issue::toRecord);
+        Set<String> uniqueAttributes = getUniqueAttributes(issues);
+        JsonArray samples = createSamples(issues, uniqueAttributes);
+        Buffer sample = Buffer.copyOf(samples.toString().getBytes());
+        JsonNode columns = guessExecutor.guessParserConfig(sample, Exec.newConfigSource(), guessConfig).getObjectNode().get("columns");
+        ConfigDiff configDiff = Exec.newConfigDiff();
+        configDiff.set("columns", columns);
+        return configDiff;
+    }
+
+    private ConfigSource createGuessConfig()
+    {
+        ConfigSource configSource = Exec.newConfigSource();
+        configSource.set("guess_plugins", new ObjectMapper().createArrayNode().add("jira"));
```

Comment:
From @tvhung83 (extracted from #54)
```suggestion
         return Exec.newConfigSource()
            .set("guess_plugins", ImmutableList.of("jira"))
            .set("guess_sample_buffer_bytes", GUESS_BUFFER_SIZE);
```

Reply:
Done